api/v1alpha1/gateway_types.go

```diff
@@ -156,7 +156,7 @@ type ListenerTLS struct {
 	//
 	// Support: Core (Kubernetes Secrets)
 	// Support: Implementation-specific (Other resource types)
-	Certificates []core.TypedLocalObjectReference `json:"certificates,omitempty"`
+	Certificates []CertificateObjectReference `json:"certificates,omitempty"`
```

Comment:
Can we assume that the referenced object is a Kubernetes object?

Reply:
Let's keep the assumption for now, we can discuss in a separate issue/PR